### ISSUE TITLE
Linux include useless shared libs?

### ISSUE THREAD

**Issue by dkdirekt** [Nuitka/Nuitka#2707]
Hello,

Today I just compiled my simple python script on OpenSuse Leap 15 and Centos 7, and I just saw the size difference between two projects, and I see 10 shared libs they are present on Centos but not on OpenSuse. So I have just tried to remove them to check if the binary work and yes, it is.

The shared libs I can remove are these:
```
libcom_err.so.2
libcrypto.so.10
libgssapi_krb5.so.2
libk5crypto.so.3
libkeyutils.so.1
libkrb5.so.3
libkrb5support.so.0
libpcre.so.1
libselinux.so.1
libssl.so.10
```

For example, libselinux.so.1 is only relative to RHEL I think...

Here is the complete file list on Centos:
```
-rw-r--r--. 1 root root   31240 10 fév 21:16 binascii.so
-rw-r--r--. 1 root root   51152 10 fév 21:16 _blake2.so
-rw-r--r--. 1 root root  153288 10 fév 21:16 _codecs_cn.so
-rw-r--r--. 1 root root  161712 10 fév 21:16 _codecs_hk.so
-rw-r--r--. 1 root root   26456 10 fév 21:16 _codecs_iso2022.so
-rw-r--r--. 1 root root  268032 10 fév 21:16 _codecs_jp.so
-rw-r--r--. 1 root root  141000 10 fév 21:16 _codecs_kr.so
-rw-r--r--. 1 root root  116424 10 fév 21:16 _codecs_tw.so
-rw-r--r--. 1 root root  119536 10 fév 21:16 _datetime.so
-rw-r--r--. 1 root root   17984 10 fév 21:16 libcom_err.so.2
-rw-r--r--. 1 root root 2644840 10 fév 21:16 libcrypto.so.10
-rw-r--r--. 1 root root  334752 10 fév 21:16 libgssapi_krb5.so.2
-rw-r--r--. 1 root root  217360 10 fév 21:16 libk5crypto.so.3
-rw-r--r--. 1 root root   18424 10 fév 21:16 libkeyutils.so.1
-rw-r--r--. 1 root root  996088 10 fév 21:16 libkrb5.so.3
-rw-r--r--. 1 root root   74272 10 fév 21:16 libkrb5support.so.0
-rw-r--r--. 1 root root  407296 10 fév 21:16 libpcre.so.1
-rw-r--r--. 1 root root 3192728 10 fév 21:16 libpython3.6m.so.1.0
-rw-r--r--. 1 root root  168592 10 fév 21:16 libselinux.so.1
-rw-r--r--. 1 root root  496440 10 fév 21:16 libssl.so.10
-rw-r--r--. 1 root root   55104 10 fév 21:16 math.so
-rw-r--r--. 1 root root   49168 10 fév 21:16 _multibytecodec.so
-rw-r--r--. 1 root root  142960 10 fév 21:16 _pickle.so
-rw-r--r--. 1 root root   27128 10 fév 21:16 _random.so
-rw-r--r--. 1 root root  105128 10 fév 21:16 _sha3.so
-rw-r--r--. 1 root root   53448 10 fév 21:16 _struct.so
-rwxr-xr-x. 1 root root 3302560 10 fév 21:16 test
-rw-r--r--. 1 root root  902736 10 fév 21:16 unicodedata.so
-rw-r--r--. 1 root root   40112 10 fév 21:16 zlib.so
```

Both installations are fresh installations only to play with Nuitka.

OS informations:
```
OS Centos 7 (fresh install only for nuitka)
kernel 3.10.0-1160.el7.x86_64
ldd (GNU libc) 2.17
gcc (GCC) 8.3.1 20190311 (Red Hat 8.3.1-3)
installed packages (yum): gcc, python3-devel, centos-release-scl, devtoolset-8-gcc, cache, patchelf
```

Nuitka informations:
```
2.0.2
    Commercial: None
    Python: 3.6.8 (default, Nov 14 2023, 16:29:52)
    Flavor: Fedora Python
    Executable: /root/test/pylibs/bin/python3
    OS: Linux
    Arch: x86_64
    Distribution: Centos (based on Fedora) 7
    Version C compiler: /usr/lib64/ccache/gcc (gcc 8).
```

python version: 
`3.6.8 (devel, native without additionnals modules installed)`

compile file:
   ```
 python3 -m venv pylibs
    source pylibs/bin/activate
    python3 -m pip install --upgrade pip
    python3 -m pip install zstandard
    python3 -m pip install nuitka
    scl enable devtoolset-8 -- python3 -m nuitka --standalone --follow-imports --output-filename=test ./test.py
```

test.py file:
`    print("hello world")`

Can you tell me if I can remove these useless libs to move the binary on other system or if I need to keep it? And if I can remove them, is there a way to remove them from compilation (unless it's a future feature/bug/other).

Thank's :)

**Comment by kayhayen** [comment]
Fedora Python (CentOS/RHEL Python) is among the worst offenders of using system dependencies that make it less portable. Compiling Python yourself will be much better also for portability. I have only really looked at Debian Python to be more minimal, but things are relatively not good there too.

As building on latest Linux is not recommended for many reasons, not the least being that the Python library is not going to like older systems a lot. You are spot on with using CentOS 7. In Nuitka commercial I have a tool to build inside an old container with optimized Python so it works on old systems too: https://nuitka.net/doc/commercial.html which does this automatically. Currently as a benefit, this also works for 3.11 as well.

This is still not described in the commercial docs, but I will add it in the coming months, also Nuitka will gain a CentOS 8 container to use with similar traits, in which you compile transparently too, but I might have to wait for 3.12 support to be solved before actually going there. Then a tier of acceptable portability will be available for everyone, and obviously, hello world will then not use too many system libraries.

**Comment by dkdirekt** [comment]
Mhh okay, I was using Centos 7 because he has libc 2.17 version so which is a good thing compared to portability. But If you recommend Debian I'll try on Debian, like Debian 9.

Thank's

**Comment by kayhayen** [comment]
I don't actually recommend Debian at all, I just said, it's the only thing I looked at for minimizing DLL intake on. I recommend self built Python in the proper way. However, that way is not really documented at this time publicly. You just need to do it on older OS for libc portability indeed. I found CentOS 7 to be a good platform for that, and expect CentOS 8 to also be one. But old Debian will of course do too. It's just important to use a virtualenv in there and no system packages.
